### DISCUSSION TITLE
How to do instancing

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3536] [Q&A] [answered]
Suppose I have an object that I want to appear at several different locations, with minimal duplication of GPU memory.  As I understand it, a renderable can only have a single transform, but I could create multiple renderables using the same `VertexBuffer`, `IndexBuffer`, and `MaterialInstance`.  Is that the way to do it?  I think the only sample that does instancing is gltf_instances, and the involvement of glTF make it a little hard for me to understand what is going on.

**Comment by romainguy** [comment, marked as answer]
Yes that's currently the cheapest way to instance a mesh. We plan on adding true GPU instancing at some point but we don't have any ETA yet (there's a feature request already opened on the tracker for this).

A Renderable is actually a cheap object, all it is is a UBO containing the transform. All actual data comes from the vertex/index buffers which can be shared. Note that you don't necessarily need a new material instance either. You can use the same material instance on multiple renderables.
